designs/native-ami-versioning.md

```diff
@@ -0,0 +1,121 @@
+
+### DriftTTL in the ProvisionerSpec
+
+Karpenter could include a drift TTL in the Provisioner. Nodes would have to be drifted for the TTL's duration to be deprovisioned. This can implement bake time, but will also apply for all future methods of Drift, such as zonal and instance permission requirements. Since other Drift methods may need more immediate actions when it occurs, setting a Drift TTL may not be granular as needed.
+
+### 🔑 Suggestion
+
+Implement Bake Time with `minimumAgeDuration`. Users can take advantage of it with EKS Optimized AMIs generally to follow requirements on AMI bake time. This applies only to AMIs, rather than a Drift TTL which would apply broadly to every provisioning requirement. Users can rely on `minimumAgeDuration` to use it as a safeguard against unexpected AMI upgrades, creating more confidence in relying on Karpenter to make the right decisions.
+
+## Release Pinning
+
+*🔑 Expand amiFamily to allow a pinned version for EKS Optimized AMIs.*
+
+Add an option to specify AMI Release Version with AMI Family. A user can pin known AMI Versions, and declaratively upgrade to a different AMI when ready. This could prevent automatic updates in the event of a newer EKS Optimized AMI. Some users [have requested this](https://github.com/aws/karpenter/issues/1495) with 13 up-votes.
+
+### *Option 1*
+
+Use an AMI Family with an appended magic string. This allows users who only know of version strings to modify AWSNodeTemplates easily to their liking.
+
+```
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+metadata:
+  name: default
+spec:
+  amiFamily: AL2-v20220824 # No version suffix defaults to latest
+```
+
+This benefits current users of EKS Optimized AMIs. It extends cleanly into defaults, and allows users an easy override that follows the same [versioning semantics](#versioning-semantics) used for each of the AMI Families, which users would be aware of.
+
+The downside is users that use custom Bootstrapping logic but do not use EKS Optimized AMIs may find this confusing if they find it in documentation or code. Confusion can be mitigated with good documentation as it’s only referring to EKS Optimized AMIs.
+
+### *Option 2*
+
+Include amiVersion as an additional filter *outside* of the amiSelector.
+
+This may be more intuitive to users that are using EKS Optimized AMIs, but has no value add for those who aren’t. This may be confusing to users looking to extend the amiVersion field into other methods of custom AMI selection, especially since it’s a top level field. One alternative is to make this mutually exclusive with AMISelector so that there’s no room for confusion.
+
+```
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+metadata:
+  name: default
+spec:
+  amiFamily: Bottlerocket
+  amiVersion: v1.9.1 # Defaults to latest, requires AMIFamily if specified
```

Comment:
I am also in favor of this, as we could further extend `amiVersion` to support [semver version ranges](https://devhints.io/semver). Some examples: 
- `1.9.1` pin to exactly this version
- `^1.9` allow for any bug-fix release upgrades
- `^1` allow for minor release upgrades
-  ...

Reply:
This would be a cool extension! For specifically the EKS Optimized AMIs, this would only work for Bottlerocket. My only qualm with this option is that we'd need to add some way to plug this into custom AMIs for it to be useful for all users.

Reply:
also `minimumAgeDuration` bake time should then probably be ignored only for a specific version (similarly like when amiID is specified) but not for a range